src/main/java/org/jabref/websocket/JabRefWebsocketServer.java

```diff
@@ -0,0 +1,476 @@
+        for (WebSocket websocket : getConnections()) {
+            WebSocketClientData webSocketClientData = websocket.getAttachment();
+            if (webSocketClientData != null && webSocketClientData.getWebSocketUID().equals(webSocketUID)) {
+                return websocket;
+            }
+        }
+
+        return null;
+    }
+
+    private boolean sendJsonString(WebSocket websocketOfRecipient, String jsonString) {
+        if (!serverStarted) {
+            return false;
+        }
+
+        if (websocketOfRecipient == null || jsonString == null) {
+            return false;
+        }
+
+        if (websocketOfRecipient.isOpen()) {
+            websocketOfRecipient.send(jsonString);
+
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public boolean sendMessage(WebSocket websocketOfRecipient, WebSocketAction webSocketAction, JsonObject messagePayload) {
+        JsonObject messageContainer = WebSocketServerUtils.createMessageContainer(webSocketAction, messagePayload);
+
+        return sendJsonString(websocketOfRecipient, new Gson().toJson(messageContainer));
+    }
+
+    public boolean sendMessage(WebSocketClientType webSocketClientTypeOfRecipient, WebSocketAction webSocketAction, JsonObject messagePayload) {
+        WebSocket websocket = getFirstWebSocketClientByWebSocketClientType(webSocketClientTypeOfRecipient);
+
+        if (websocket != null) {
+            return sendMessage(websocket, webSocketAction, messagePayload);
+        }
+
+        return false;
+    }
+
+    public boolean sendMessage(String webSocketUIDofRecipient, WebSocketAction webSocketAction, JsonObject messagePayload) {
+        WebSocket websocket = getWebSocketClientByWebSocketUID(webSocketUIDofRecipient);
+
+        if (websocket != null) {
+            return sendMessage(websocket, webSocketAction, messagePayload);
+        }
+
+        return false;
+    }
+
+    public void broadcastMessage(WebSocketAction webSocketAction, JsonObject messagePayload) {
+        for (WebSocket websocket : getConnections()) {
+            sendMessage(websocket, webSocketAction, messagePayload);
+        }
+    }
+
+    public boolean startServer() {
+        if (serverStarting) {
+            LOGGER.info("[ws] JabRefWebsocketServer is already starting");
+
+            return false;
+        } else if (serverStarted) {
+            LOGGER.info("[ws] JabRefWebsocketServer has already been started");
+
+            return false;
+        } else {
+            LOGGER.info("[ws] JabRefWebsocketServer is starting up...");
+
+            serverStarting = true;
+
+            addShutdownHook();
+            setConnectionLostTimeout(connectionLostTimeoutValue);
+            start();
+
+            return true;
+        }
+    }
+
+    public boolean stopServer() {
+        if (serverStarting) {
+            LOGGER.info("[ws] JabRefWebsocketServer is currently starting up and cannot be stopped during this process");
```

Comment:
I think most of these logger statements can be degraded to `debug`. Info messages are shown to the user in the Console Log.